server/routes/events.js

```diff
@@ -1,15 +1,258 @@
+router.get('/filter', function(req, res, next) {
+  const date = req.query.date;
+  const month = req.query.month;
+  const year = req.query.year;
+  const quarter = req.query.quarter;
+  const start_date = req.query.start_date;
+  const end_date = req.query.end_date;
+  const category = req.query.category;
+  const company = req.query.company;
+  const featured = req.query.featured;
+  const past = req.query.past;
+
+  // example: 2018-10-19
+  if (date != undefined)
+  {
+  	db.query('SELECT fb_id FROM event WHERE date(starts_at) <= \'' + date + '\' AND \'' + date + '\' <= date(ends_at)',[], (err, result) => {
+    	if (err) {
+      	  return next(err);
+    	}
+    	res.send(result.rows);
+ 	});
+  } else if (month != undefined) {
+  	db.query('SELECT * FROM event WHERE EXTRACT(MONTH FROM starts_at) = \'' + month +'\' OR EXTRACT(MONTH FROM ends_at) = \'' + month + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	});
+  } else if (year != undefined) {
+  	db.query('SELECT * FROM event WHERE EXTRACT(YEAR FROM starts_at) = \'' + year + '\' OR EXTRACT(YEAR FROM ends_at) = \'' + year + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (quarter != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE quarter = \'' + quarter + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (start_date != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE date(starts_at) = \'' + start_date + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (end_date != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE date(starts_at) = \'' + end_date + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (category != undefined) {
+  	db.query('SELECT event_id FROM event_category WHERE category_id = \'' + category_id + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (company != undefined) {
+  	db.query('SELECT event_id FROM event_company WHERE company_id = \'' + company + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (featured != undefined) {
+  	db.query('SELECT * FROM event WHERE is_featured = \'' + featured + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (past != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE ends_at < now()', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else {
+  	res.json({result: false}); // error message
+  }
 });
 
-// GET events listing. 
+// Get all events
```

Comment:
Just a general comment, usually for the HTTP requests (GET, POST, PUT, DELETE, etc.) I caps each occurrence to clarify what HTTP method is being used. e.g. "GET all events" rather than "Get all events"